#!/bin/sh -e
##  by Arnaud Cornet <arnaud.cornet@gmail.com>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fixes locale handling in case there is a trailing @\w+ at the
## DP: end of the locale name (like in fr_FR@euro)

if [ $# -lt 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch} ${2:+-d $2}"

case "$1" in
       -patch) patch $patch_opts -p1 < $0;;
       -unpatch) patch $patch_opts -p1 -R < $0;;
        *)
                echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
                exit 1;;
esac

exit 0

diff -puriN zonecheck-2.0.4.orig/zc/param.rb zonecheck-2.0.4/zc/param.rb
--- zonecheck-2.0.4.orig/zc/param.rb	2007-04-06 11:58:50.000000000 +0200
+++ zonecheck-2.0.4/zc/param.rb	2007-04-06 11:59:08.000000000 +0200
@@ -523,7 +523,7 @@ class Param
 	end
 
 	def local=(resolv)
-	    resolv = resolv.clone.untaint if resolv.tainted?
+	    resolv = resolv.dup.untaint if resolv.tainted?
 	    @local_name = if resolv.nil? || resolv =~ /^\s*$/
 			  then nil
 			  else resolv
